### DISCUSSION TITLE
[Question]: Error when running Test Next Major action

### DISCUSSION THREAD

**Discussion by fvet** [microsoft/AL-Go#1588] [Q&A]
### Question

We have the following setup:

One repo, with multiple projects / apps, most having dependencies on each other (minimum against Core)

![image](https://github.com/user-attachments/assets/6aced8ec-2340-47f9-b611-484207e6579a)

Core app has an additional dependency against a 3rd party app, which is included in the repo under Core project, .appSourceCopPackages folder.

![image](https://github.com/user-attachments/assets/09191609-2d36-475b-b973-d6554db4e3b7)

Core app manifest file:

```
{
  "id": "c1fa90da-fc3f-4720-abd8-1b30fe8f3faf",         
  "name": "ESC Core",
  "publisher": "ESC bvba",
  "version": "24.0.0.0",
  "brief": "ESC Core Module",
  "description": "ESC Core module",
  "privacyStatement": "https://www.esc.be/en/privacy-policy",
  "EULA": "https://www.esc.be/en/terms-and-conditions",
  "help": "https://esc.be/support/",
  "url": "http://www.esc.be",
  "logo": "ESC.png",
  "screenshots": [],
  "platform": "24.0.0.0", // <<<
  "application": "24.0.0.0",  // <<<
  "idRanges": [
    {
      "from": 71791569,
      "to": 71791574
    }
  ],  
  "dependencies": [
    {
      "id": "dbe4cf4c-5910-4ad1-beee-394010c59802",
      "name": "Dynavision Core",
      "publisher": "Dynavision",
      "version": "23.0.14.0"
    }
  ],
  "contextSensitiveHelpUrl": "https://esc.be/",
  "runtime": "13.0",  // <<<
  "target": "Cloud",
```

All actions run well / compile properly against BC24. 
We would like to be informed about possible AppSource validation errors, by periodically running the  'Text Next Major' action on the main branch.

When running the 'Text Next Major' action for the entire repo, the action fails with errors:

```
Error: AL1153 The referenced module 'System by Microsoft (25.0.24293.0)' with runtime reference version '14.0' cannot be loaded by the compiler with version '13.0'.
Error: AL1153 The referenced module 'System Application by Microsoft (25.0.23364.24332)' with runtime reference version '14.0' cannot be loaded by the compiler with version '13.0'.
Error: AL1153 The referenced module 'Business Foundation by Microsoft (25.0.23364.24332)' with runtime reference version '14.0' cannot be loaded by the compiler with version '13.0'.
Error: AL1153 The referenced module 'Base Application by Microsoft (25.0.23364.24332)' with runtime reference version '14.0' cannot be loaded by the compiler with version '13.0'.
Error: AL1153 The referenced module 'Application by Microsoft (25.0.23364.24332)' with runtime reference version '14.0' cannot be loaded by the compiler with version '13.0'.

```
[logs_28540920193.zip](https://github.com/user-attachments/files/17058933/logs_28540920193.zip)

_Side note: As mentioned on Yammer (https://www.yammer.com/dynamicsnavdev/#/threads/show?threadId=2959605473320960), our devs are focusing on the maintaining the main branch only (no plans to increase the application version to 25.0 within the next 5 months), how can they be alerted of these 25.0 validation errors upfront when working on a branch that has minimum dependency set to 24.0? (while avoiding the hassle of maintaining separate 'nextmajor' branches or uptaking preprocessorsymbols). Their reality is still 24.0._
...
_we're not looking to implement the 'hotfix' process (https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-hotfixing-appsource-app) , as we won't release any Application 25.0 version soon._

**Comment by frottke** [comment]
We are getting the same error, were you able to find the source of the problem?

**Comment by fvet** [comment]
@frottke We haven't done any further investigation, awaiting feedback from MS on how to manage / configure.

**Comment by waldo1001** [comment]
Judging from the error - did you try to use the Prerelease-compiler?

**Comment by fvet** [comment]
I'm not aware of an AL:GO setting for using the Prerelease-compiler? Couldn't find any proper guidelines on how to achieve this in the Usage Scenarios / Workshop / Settings documentation...

**Comment by freddydk** [comment]
The problem here is the missing partner runtime package.
When running Test Next Major - it automatically uses the compiler from the Next Major artifacts.
I will do some investigations and get back in a few days

**Comment by frottke** [comment]
> The problem here is the missing partner runtime package. When running Test Next Major - it automatically uses the compiler from the Next Major artifacts. I will do some investigations and get back in a few days

I'm not exactly sure what you mean with missing partner runtime packages. We have in our case no dependencies to other apps except the Microsoft Application.

the error also appears when using localdevenv.ps1 (might be easier to debug? :) )

**Comment by freddydk** [comment]
Did this issue get resolved after the release of 25.0 ?
Does Test Next Major work with v26 now?

**Comment by fvet** [comment]
Still failing :(

[logs_29733413758.zip](https://github.com/user-attachments/files/17408689/logs_29733413758.zip)

**Comment by freddydk** [comment]
Could you create a public repository, which reproes this problem?
My bingmaps repo: https://github.com/microsoft/bcsamples-bingmaps.pte - works fine with nextmajor (although it fails some tests - but building works)

**Comment by frottke** [comment]
We are also getting the same error with 24.0 (26.0 next major). I can provide a public repository, might be not today though.

**Comment by rcfoyn** [comment]
More than 8 months later, this issue (converted to discussion without explicit justification?) is still Unanswered. Please provide an update, thank you.

**Reply by aholstrup1** [reply]
The issue was originally posted as a "Question". A while back we converted all issues with the label "Question" into discussions. 

If you could share the logs for your failing build that would be helpful. Preferably also a public repository that reproduces the problem.
